File: SatinalmaPro.Android/app/src/main/java/com/satinalmapro/android/core/helpers/OnayBildirimYardimcisi.kt
package com.satinalmapro.android.core.helpers

import com.satinalmapro.android.core.roles.KullaniciRolleri

/** Masaüstü OnayBildirimYardimcisi ile aynı hedefleme mantığı. */
object OnayBildirimYardimcisi {
    fun satinalmaOnayladi(onaylayanRol: String?): Boolean =
        KullaniciRolleri.normalize(onaylayanRol) == KullaniciRolleri.SATINALMA

    fun onaylandiHedefleri(
        olusturanUid: String?,
        onaylayanRol: String?
    ): List<Pair<String?, String?>> {
        val hedefler = mutableListOf<Pair<String?, String?>>()
        if (satinalmaOnayladi(onaylayanRol)) {
            if (!olusturanUid.isNullOrBlank()) hedefler.add(null to olusturanUid)
            hedefler.add(KullaniciRolleri.YONETIM to null)
            return hedefler
        }
        if (!olusturanUid.isNullOrBlank()) hedefler.add(null to olusturanUid)
        hedefler.add(KullaniciRolleri.SATINALMA to null)
        return hedefler
    }

    fun teklifIstemeBildirimEk(onaylayanRol: String?): String =
        if (satinalmaOnayladi(onaylayanRol)) {
            "Satınalma birimi teklifiniz için teklif girişi talep etti."
        } else {
            "Yönetim teklifiniz için satınalmadan teklif talep etti."
        }
}

File: SatinalmaPro.Shared/Helpers/OnayBildirimYardimcisi.cs
using SatinalmaPro.Shared.Models;

namespace SatinalmaPro.Shared.Helpers;

public static class OnayBildirimYardimcisi
{
    public static bool SatinalmaOnayladi(string? onaylayanRol) =>
        KullaniciRolleri.Normalize(onaylayanRol) == KullaniciRolleri.Satinalma;

    public static IEnumerable<(string? HedefRol, string? HedefUid)> OnaylandiHedefleri(
        string? olusturanUid,
        string? onaylayanRol)
    {
        if (SatinalmaOnayladi(onaylayanRol))
        {
            if (!string.IsNullOrWhiteSpace(olusturanUid))
                yield return (null, olusturanUid);

            yield break;
        }

        if (!string.IsNullOrWhiteSpace(olusturanUid))
            yield return (null, olusturanUid);

        yield return (KullaniciRolleri.Satinalma, null);
    }

    public static string TeklifIstemeBildirimEk(string? onaylayanRol) =>
        SatinalmaOnayladi(onaylayanRol)
            ? "Satınalma birimi teklifiniz için teklif girişi talep etti."
            : "Yönetim teklifiniz için satınalmadan teklif talep etti.";
}
